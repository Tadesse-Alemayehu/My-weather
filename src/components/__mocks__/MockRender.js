import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/configureStore';

export default (component) => (
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>
);
