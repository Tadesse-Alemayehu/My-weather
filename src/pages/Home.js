import React from 'react'
import { Container, Row, Col } from 'react-bootstrap';
import { useSelector } from 'react-redux'
import Weather from '../components/Weather';
import FilteredCities from '../components/FilteredCities';
import './home.scss'
export default function Home() {
  const cities = useSelector(store => store.cities);
  const filterType = useSelector(store => store.filter);
  const myCity = cities.filter(city => city.nearMe === 0);
  const cityNearMe = cities.filter(city => city.nearMe === 1);
  const otherCities = cities.filter(city => city.nearMe === 2);
  cities.forEach(city => console.log(city.weather[0].description))
  return (
    (filterType === 'All') && < Container >
      <h3 className="title">My cities</h3>
      <Row >
        {myCity.length > 0 && myCity.map(city => <Weather key={city.id} city={city} />) || <p>You havent checked any city yet</p>}
      </Row>
      <h3 className="title">Cities near you</h3>
      <Row >
        {cityNearMe.length > 0 && cityNearMe.map(city => <Weather key={city.id} city={city} />) || <p>We couldn't find your location</p>}
      </Row>
      <h3 className="title">Other cities</h3>
      <Row >
        {otherCities.length > 0 && otherCities.map(city => <Weather key={city.id} city={city} />) || <p>You haven't checked other cities</p>}
      </Row>
    </ Container> || <FilteredCities filter={filterType} cities={cities} />
  )
}
