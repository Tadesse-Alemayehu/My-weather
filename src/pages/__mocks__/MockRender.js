import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/configureStore';

const MockRender = (component) => (
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>
);

export default MockRender;
