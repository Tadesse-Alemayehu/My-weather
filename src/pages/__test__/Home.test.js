import { render, screen } from '@testing-library/react'
import Home from '../Home';
import '@testing-library/jest-dom';
import MockRender from '../__mocks__/MockRender'
import renderer from 'react-test-renderer'
describe('Home page test', () => {
  test('Should display the error message', () => {
    render(MockRender(<Home />));
    const headings = screen.getAllByRole(/heading/i);
    expect(headings.length >= 3).toBeTruthy();
  })
});


describe('Snapshot check for the Home page', () => {

  test('should match the snapshot', () => {
    const homePageSnap = renderer
      .create(MockRender(<Home />)).toJSON();
    expect(homePageSnap).toMatchSnapshot();
  })
})