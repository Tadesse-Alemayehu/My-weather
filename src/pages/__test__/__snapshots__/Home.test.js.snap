// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot check for the Home page should match the snapshot 1`] = `
Array [
  <h3
    className="title"
  >
    My cities
  </h3>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <p
        className="warning"
      >
        You haven't checked any city yet
      </p>
    </div>
    <hr />
  </div>,
  <h3
    className="title"
  >
    Cities near you
  </h3>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <p
        className="warning"
      >
        We couldn't find your location
        <br />
         
        make sure to allow a location
      </p>
    </div>
    <hr />
  </div>,
  <h3
    className="title"
  >
    Other cities
  </h3>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <p
        className="warning"
      >
        You haven't checked other cities
      </p>
    </div>
  </div>,
]
`;
